### PR TITLE
core: Remove unused backtracking function in parser

### PR DESCRIPTION
**Pull request by math-fehr**
This function was unused, and was from the old lexer.

### CODE REVIEW

**Comment by math-fehr** [comment]
This also means we can remove the deferred exceptions, as they were only used because of the backtracking.
I benched it on a large file, and I saw no changes in runtime.

File: xdsl/utils/exceptions.py [hunk 1]
```diff
@@ -65,97 +65,27 @@ def __str__(self) -> str:
...
-        refs: list[tuple[Span, str | None]],
-        history: "BacktrackingHistory" | None = None,
-    ):
-        super(MultipleSpansParseError, self).__init__(span, msg, history)
-        self.refs = refs
-        self.ref_text = ref_text
-
-    def print_pretty(self, file: IO[str] = sys.stderr):
-        super(MultipleSpansParseError, self).print_pretty(file)
+    def print(self, file: IO[str] = sys.stderr):
```
Comment (superlopuh) [inline xdsl/utils/exceptions.py:87]: do you need this?

**Review by superlopuh — APPROVED** [review]

File: xdsl/utils/exceptions.py (hunk 1 above)
Comment (math-fehr) [reply xdsl/utils/exceptions.py:87]: Yes, this is what is called from the parent class!

Comment (math-fehr) [reply xdsl/utils/exceptions.py:87]: Actually, let me remove that and just keep __repr__

**Pull request merged**